api/v0/ingest/schema/types.go

```diff
@@ -0,0 +1,178 @@
+package schema
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/node/bindnode"
+	"github.com/multiformats/go-multihash"
+)
+
+type (
+	Advertisement struct {
+		PreviousID *ipld.Link
+		Provider   string
+		Addresses  []string
+		Signature  []byte
+		Entries    ipld.Link
+		ContextID  []byte
+		Metadata   []byte
+		IsRm       bool
+	}
+	EntryChunk struct {
+		Entries []multihash.Multihash
+		Next    *ipld.Link
```

Comment:
I know this pointer to interface feels weird, because an interface can already be nil, which should be enough for optional/nullable. See https://github.com/ipld/go-ipld-prime/issues/378 FYI

Reply:
Great to see that issue captured 👍 In the meantime, I have to use pointer to interfaces right?

Reply:
Yup.